hbase-server/src/main/java/org/apache/hadoop/hbase/regionserver/RSRpcServices.java

```diff
@@ -418,6 +418,18 @@ public void run() {
     }
   }
 
+  static class QuotaAndRegionScannerHolder {
+    private final String scannerName;
+    private final RegionScannerHolder rsh;
+    private final OperationQuota quota;
+
+    QuotaAndRegionScannerHolder(String scannerName, RegionScannerHolder rsh, OperationQuota quota) {
```

Comment:
Instead of wrapping the existing holder class, does it make sense to push the Quota and name information down into the `RegionScannerHolder`? I imagine that there's other related places where things could be cleaner if the Quota was automagically along for the ride. Maybe rename it to a `Context` objects instead of "holder".

Reply:
Good callout, will refactor here

Reply:
I think this makes sense, I've refactored the code a bit here.

Reply:
The only oddity here is that now the context may hold a Quota that isn't up-to-date because updating the RegionScannerContext's quota isn't thread-safe. There are other variables of the RSX that are updated this way though, so maybe this isn't an actual issue.

Reply:
That doesn't sound great. Maybe this is why the quota isn't already in that context object.

Reply:
I think I can make this an `AtomicReference<OperationQuota>` to overcome this issue

Reply:
I think even if we make update the context to hold an `AtomicReference<OperationQuota>`, it's a bit odd for there to be two sources of truth for the quota. The first being the RpcQuotaManager and the second being the cached scanner context. 

I'm going to roll this change back because I think it adds more complexities and isn't solving any pertinent issues at the moment